Yumemi-ios-trainingTests/Yumemi_ios_trainingTests.swift

```diff
@@ -9,28 +9,91 @@ import XCTest
+    
+    func test_天気予報がcloudyだったら_画面に曇り画像が表示されること() {
+        let infraWeatherInfo = InfraWeatherInfo(maxTemp: 10, minTemp: 5, date: Date(), weather: .cloudy)
+        let weatherInfo = WeatherInfoConverter.convert(infraWeatherInfo: infraWeatherInfo)
+        let weatherUIData = WeatherUIData(weatherInfo: weatherInfo)
+        
+        let exp = self.getExpectation(#function)
+        presenter.exp = exp
+        weatherClient.dummyInfraWeatherInfo = infraWeatherInfo
+        vc.onTapReloadButton("")
+        
+        wait(for: [exp], timeout: 5.0)
+        XCTAssertEqual(vc.weatherImageView.image!, weatherUIData.image)
     }
-
-    func testPerformanceExample() throws {
-        // This is an example of a performance test case.
-        self.measure {
-            // Put the code you want to measure the time of here.
-        }
+    
+    func test_天気予報がrainyだったら_画面に雨画像が表示されること() {
+        let infraWeatherInfo = InfraWeatherInfo(maxTemp: 10, minTemp: 5, date: Date(), weather: .rainy)
+        let weatherInfo = WeatherInfoConverter.convert(infraWeatherInfo: infraWeatherInfo)
+        let weatherUIData = WeatherUIData(weatherInfo: weatherInfo)
+        
+        let exp = self.getExpectation(#function)
+        presenter.exp = exp
+        weatherClient.dummyInfraWeatherInfo = infraWeatherInfo
+        vc.onTapReloadButton("")
+        
+        wait(for: [exp], timeout: 5.0)
+        XCTAssertEqual(vc.weatherImageView.image!, weatherUIData.image)
+    }
+    
+    func test_天気予報の最高気温がUILabelに反映されること() {
+        let infraWeatherInfo = InfraWeatherInfo(maxTemp: 10, minTemp: 5, date: Date(), weather: .rainy)
+        let weatherInfo = WeatherInfoConverter.convert(infraWeatherInfo: infraWeatherInfo)
+        let weatherUIData = WeatherUIData(weatherInfo: weatherInfo)
+        
+        let exp = self.getExpectation(#function)
+        presenter.exp = exp
+        weatherClient.dummyInfraWeatherInfo = infraWeatherInfo
+        vc.onTapReloadButton("")
+        
+        wait(for: [exp], timeout: 5.0)
+        XCTAssertEqual(vc.maxTempLabel.text!, weatherUIData.maxTemp)
+    }
+    
+    func test_天気予報の最低気温がUILabelに反映されること() {
+        let infraWeatherInfo = InfraWeatherInfo(maxTemp: 10, minTemp: 5, date: Date(), weather: .rainy)
+        let weatherInfo = WeatherInfoConverter.convert(infraWeatherInfo: infraWeatherInfo)
+        let weatherUIData = WeatherUIData(weatherInfo: weatherInfo)
+        
+        let exp = self.getExpectation(#function)
+        presenter.exp = exp
+        weatherClient.dummyInfraWeatherInfo = infraWeatherInfo
+        vc.onTapReloadButton("")
+        
+        wait(for: [exp], timeout: 5.0)
+        XCTAssertEqual(vc.minTempLabel.text!, weatherUIData.minTemp)
+    }
+    
+    private func getExpectation(_ funcName: String) -> XCTestExpectation {
+        XCTestExpectation(description: "\(funcName)のタイムアウト")
```

Comment:
[nits]
```suggestion
    private func makeExpectation(_ funcName: String) -> XCTestExpectation {
        XCTestExpectation(description: "\(funcName)のタイムアウト")
```

のほうが良さそうです。

Reply:
get の場合、既に生成されているものを取得することを期待し、
make の場合、何か生成されることを期待します。